### DISCUSSION TITLE
Feature request - include build information

### DISCUSSION THREAD

**Discussion by dallemang** [edmcouncil/ontology-publisher#106] [Ideas]
The published files prod.fibo.quickstart and dev.fibo.quickstart do not include the build version information (e.g., 2021Q3).  As far as I can tell, this information isn't available in a queryable form anywhere in any publication, nor in github; it is only available as a part of a versionIRI in an import. 

If someone downloads a version of FIBO, and saves it elsewhere, it is difficult to determine which version it is.  

Can we include that information in the published files somehow?  Maybe by including some of the data from the file https://github.com/edmcouncil/fibo/blob/master/MetadataFIBO.rdf , and then adding the version information, perhaps using dct:hasVersion ?

**Comment by mereolog** [comment]
We will update https://github.com/edmcouncil/ontology-publisher as soon as possible.

**Comment by mereolog** [comment]
Will adding something like this be enough?

```
@prefix fibo-spec: <https://spec.edmcouncil.org/fibo/ontology/MetadataFIBO/> .
fibo-spec:  owl:versionIRI  <https://spec.edmcouncil.org/fibo/ontology/20201201/MetadataFIBO/> .
```

**Comment by dallemang** [comment]
I'd like the release string as well .... eg 2021Q3

**Comment by mereolog** [comment]
@dallemang Fair enough, but at the moment we do not store this kind of release information within FIBO ontologies, i.e., it is sort of external.
On the other hand, we can add quite easily the timestamp when quick FIBO versions were created, well perhaps when the ontology-publisher process started.

**Comment by rivettp** [comment]
It should be in the manifest each time we publish from Dev or Prod - we already have the long-standing issue for this in ontology-publisher.

**Comment by mereolog** [comment]
@rivettp what do you mean by the "manifest"? Something like the manifest file in JAVA?

**Comment by rivettp** [comment]
@mereolog yes a separate file. We already produce one but it's useless. We already have an issue for this we've discussed in Jira https://jira.edmcouncil.org/browse/INFRA-264?jql=text%20~%20%22manifest%22

**Comment by rivettp** [comment]
This should really be an ontology publisher issue

**Comment by mereolog** [comment]
@rivettp you are right - I transferred it here.

**Comment by mereolog** [comment]
What about having the manifest that looks like this:
```
@prefix : <https://spec.edmcouncil.org/fibo/ontology/f5e59fdf216e352428762668ac1f4178daff933b#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@base <https://spec.edmcouncil.org/fibo/ontology/f5e59fdf216e352428762668ac1f4178daff933b> .

<https://spec.edmcouncil.org/fibo/ontology/f5e59fdf216e352428762668ac1f4178daff933b> rdf:type owl:Ontology ;
                                                                                      owl:imports <http://purl.org/dc/terms> ;
                                                                                      <http://purl.org/dc/elements/1.1/date> "Thu Sep 2 10:57:33 2021 +0100" ;
                                                                                      <http://purl.org/dc/elements/1.1/identifier> "f5e59fdf216e352428762668ac1f4178daff933b" .
```
f5e59fdf216e352428762668ac1f4178daff933b is the commit id.

**Comment by rivettp** [comment]
Not sure this should be an owl:Ontology. It's really a (build) Package. SPDX has an ontology for this sort of thing https://github.com/spdx/spdx-spec/tree/development/v2.2.1/ontology though I would not suggest using any more than the basics of it. But having looked at it reminds me we should follow best practice and have checksums for the downloads (probably a separate issue).

It should be both machine readable and make sense to a business user who just wants to check they have the build that fixes an issue they're interested in. Note that not all users even have access to GitHub from their corporate environment.
Ideally there should be a GitHub URL to the pull request, rather than leaving people to figure out how to deconstruct a cryptic URI. Better still if we can additionally have the PR summary and links to GitHub issues fixed.

**Comment by mereolog** [comment]
Ok, thank for the reference.
We may create an instance of http://spdx.org/rdf/terms#Package for each PR, but I do not get what IRI you want to give to this instance. Could you clarify?
